R/price_complete.R

```diff
@@ -0,0 +1,53 @@
+#' Interpolate prices for a specific product within a given date range.
+#'
+#' @param product_id A product ID.
+#' @param date_min The minimum date for the interpolation range as a character string in the format "YYYY-mm-dd".  Defaults to the minimum date available for the specified product ID.
+#' @param date_max The maximum date for the interpolation range as a character string in the format "YYYY-mm-dd".  Defaults to the maximum date available for the specified product ID.
+#'
+#' @return Interpolated prices for the specified (or default) date range as a \code{data.frame}.
+#' @export
+#'
+#' @examples
+#' # Interpolate prices for a specific product within a given date range.
+#' \dontrun{
+#' price_complete(product_id = 1, date_min = "2020-01-01", date_max = "2020-12-31")
+#' }
+price_complete <- function(product_id, date_min = NULL, date_max = NULL) {
+
+  if (!is.null(date_min)) {
+    date_min_check <- try(as.Date(date_min, format="%Y-%m-%d"))
```

Comment:
You end up calling `as.Date()` twice for each date. Surely you could set `date_min` directly here?

Same applies for `date_max`.